File: Semestrul 4/MPP/Proiecte/java_gui/src/main/java/map/service/DonatorService.java
package map.service;

import map.domain.Donator;
import map.repository.database.DonatorDBRepo;
import map.repository.database.DonatorRepo0;

import java.util.List;

public class DonatorService {
    private DonatorRepo0 donatorDBRepo;

    public DonatorService(DonatorDBRepo donatorDBRepo) {
        this.donatorDBRepo = donatorDBRepo;
    }

    public List<Donator> searchByName(String partialName){
        return donatorDBRepo.findByNameContaining(partialName);
    }

    public Donator findById(Integer id) {
        return null;
    }

    public void save(Donator donator) {
        donatorDBRepo.save(donator);
    }

    public Donator findByFullName(String numeDonator) {
        return donatorDBRepo.findByFullName(numeDonator);
    }
}

File: Semestrul 4/MPP/Proiecte/c#_gui/service/DonatorService.cs
using System.Collections.Generic;
using csharp_avalonia.domain;
using csharp_avalonia.repository;

namespace csharp_avalonia.service;

public class DonatorService
{
    private readonly DonatorRepo0 donatorRepo;

    public DonatorService(DonatorRepo0 donatorRepo1)
    {
        donatorRepo = donatorRepo1;
    }

    public List<Donator> SearchByName(string partialName)
    {
        return donatorRepo.FindByNameContaining(partialName);
    }

    public Donator FindByFullName(string donorName)
    {
        return donatorRepo.FindByFullName(donorName);
    }

    public Donator save(Donator donator)
    {
        return donatorRepo.save(donator);
    }
}

